src/stateChangeCapture.ts

```diff
@@ -8,29 +8,36 @@ export interface LastItemState {
 }
 
 export function sanitizePipelineName(pipelineName: string | undefined): string {
-
   const sanitisePatterns = process.env.SANITISE_PATTERNS || "-pipeline"
   const removeWhitespace = (x: string) => x.trim()
 
   if (pipelineName) {
     const sanitise = pattern => {
-      const matchResult = pipelineName.split(pattern)
-      if (matchResult.length > 1) {
-        return matchResult[0]
-      } else {
-        //no match
-        return undefined
+      if (process.env.IS_REGEX && process.env.IS_REGEX == "true") {
```

Comment:
If you're adding a new environment variable, this needs to be exposed in the template.

Ideally, I'd prefer to be able to detect regex on the fly (and potentially mix and match patterns)

Reply:
How would you recommend we do it while keeping backward compatibility?
I was actually hesitant to introuduce an env variable and i did it purely to maintain backward compatibility.
For instance, -pipeline is also a valid regex but if i were to pass -pipeline as a pattern to sanitize, The code will get the string before "-pipeline". 
Whereas when i am passing a regex pattern - I am trying to match the regex pattern itself and not what comes before the match. Is it still the functionality that we are aiming for? Would it be okay to remove the previous functionality and do pure regex.
The only implication i can think of is that people will have to create alarm names matching the sanitized pipeline name, which would become the actual pattern they are matching and not what would come before "-pipeline" or whatever we pass in.